tests/post/steps/test_volume.py

```diff
@@ -31,13 +32,32 @@ def test_deploy_operator(host):
 def test_volume_creation(host):
     pass
 
+@scenario('../features/volume.feature', 'Test volume deletion (sparseLoopDevice)')
+def test_volume_deletion(host):
+    pass
+
+# }}}
+
+# Given {{{
+
+@given(parsers.parse("a Volume '{name}' exists"))
+def volume_exist(host, name, k8s_custom_client):
+    volume = _volume_exist(host, name, k8s_custom_client)
+    if volume is None:
+        _create_volume(host, name, k8s_custom_client)
```

Comment:
Here you should have:

```python
if volume is None:
    body = DEFAULT_VOLUME.format(name=name)
    _create_volume(k8s_custom_client, body)
```

Reply:
We also probably need to have a wait here, until the Volume becomes `Available`.

Reply:
I fail to see why we need to wait here. We are making a request to the API and once the request is submitted, we should terminate the transaction.

Checking the Availability of the created volume should be handled in the next stage which is what we do actually.

Reply:
Nope, the role of the `Given` is to ensure that the volume exists and thus should wait for it to becomes available.

The test scenario is:

```cucumber
Scenario: Test volume deletion (sparseLoopDevice)
    Given a Volume 'volume2' exists
    When I delete the Volume 'volume2'
    Then the Volume 'volume2' does not exists
    And the PersistentVolume 'volume2' does not exists
```

We try to delete right after we are sure that the Volume exists: there is no explicit `wait` to add in the scenario because it must be handled by the given.